src/main/java/bio/terra/service/dataset/flight/update/DatasetSchemaUpdateValidateModelStep.java

```diff
@@ -37,16 +39,25 @@ public StepResult doStep(FlightContext context) throws InterruptedException, Ret
     Collection<String> uniqueTableNames =
         CollectionUtils.intersection(existingTableNames, newTableNames);
 
-    if (uniqueTableNames.isEmpty()) {
-      return StepResult.getStepResultSuccess();
+    if (!uniqueTableNames.isEmpty()) {
+      return new StepResult(
+          StepStatus.STEP_RESULT_FAILURE_FATAL,
+          new DatasetSchemaUpdateException(
+              "Could not validate table additions",
+              List.of(
+                  "Found new tables that would overwrite existing tables",
+                  String.join(", ", uniqueTableNames))));
     }
-    return new StepResult(
-        StepStatus.STEP_RESULT_FAILURE_FATAL,
-        new DatasetSchemaUpdateException(
-            "Could not validate table additions",
-            List.of(
-                "Found new tables that would overwrite existing tables",
-                String.join(", ", uniqueTableNames))));
+    if (newTableNames.stream().distinct().count() != newTableNames.size()) {
+      String duplicateTables = newTableNames.stream().collect(Collectors.groupingBy(Function.identity(), Collectors.counting())).entrySet().stream().filter(e -> e.getValue() > 1L).map(Map.Entry::getKey).collect(Collectors.joining(", "));
```

Comment:
Yikes, this would be easier to read if you at least put it across different lines

Reply:
Also you're doing this complicated thing just for messaging...I'd actually do it outside of the if statement and then make the if statement check if duplicate tables (maybe make it a `Set`) is empty or not

Reply:
I think you're reviewing an old revision. I agree that the logic should be moved out of the if statement, but your comment is on the wrong class!